app/views/cart/show.html.erb

```diff
@@ -37,18 +37,25 @@
           </p>
           <p><%= link_to "Remove", "/cart/#{item.id}", method: :delete %></p>
         </center>
-        </td>
+      </td>
       <td>
         <p><%=number_to_currency(cart.subtotal(item))%></p>
-    </td>
+      </td>
     </tr>
-  <% end %>
-</table>
-  <p>Total: <%= number_to_currency(cart.total) %></p>
-  <p><%= link_to "Empty Cart", "/cart", method: :delete %></p>
-  <p><%= link_to "Checkout", "/orders/new", method: :get %></p>
-<% else %>
-  <h3 align= "center">Cart is currently empty</h3>
-<% end %>
-<table>
+    <% end %>
+  </table>
+  <section id="cart-functions">
+    <p>Total: <%= number_to_currency(cart.total) %></p>
+    <p><%= link_to "Empty Cart", "/cart", method: :delete %></p>
+    <% if !current_user %>
+    <p> You need to register or login to checkout </p>
+    <p> <%= link_to "log in", "/login" %> </p>
+    <p> <%= link_to "register", "/register" %> </p>
+    <% else %>
```

Comment:
US 25 says that the words "register" and "login" in the message should be links. Try:
`<p>You need to <%= link_to "register", "/register" %> or <%= link_to "log in", "/login" %> to checkout</p>`

Reply:
That worked!

Reply:
Update PR with what Alex said